### PR TITLE
Properly set cmd and entrypoint to null when needed.

### PR DESCRIPTION
**Pull request by xingao267**
If cmd or entrypoint is set to one of {None, [], ""}, it will be set to null in the image config.

#379

### CODE REVIEW

**Comment by siddharthab** [comment]
Thank You!

File: container/image.bzl [hunk 1]
```diff
@@ -380,6 +394,13 @@ _attrs = dict(_layer.attrs.items() + {
         executable = True,
         allow_files = True,
     ),
+    # null_cmd and null_entrypoint are hidden attributes from users.
```
Comment (nlopezgi) [inline container/image.bzl]: You need to document somewhere much more visible that explicitly setting cmd or entrypoint to None, [] or "" will have this behavior (here: https://github.com/bazelbuild/rules_docker#container_image-1 , https://github.com/bazelbuild/rules_docker/blob/master/container/image.bzl#L476,  https://github.com/bazelbuild/rules_docker/blob/master/container/image.bzl#L482)

File: tests/docker/BUILD [hunk 2]
```diff
@@ -39,3 +39,53 @@ container_test(
...
+)
+
+container_image(
+    name = "image_with_null_cmd_entrypoint",
+    base = ":image_with_some_cmd_entrypoint",
+    cmd = [],
+    entrypoint = None,
+)
+
+container_test(
```
Comment (nlopezgi) [inline tests/docker/BUILD]: What happens if you have another container_image that uses image_with_null_cmd_entrypoint as base but sets a different (not null) entrypoint (i.e., can you verify the topmost layer overrides everything below)?

set_cmd_and_entrypoint_null_on_base

File: tests/docker/BUILD [hunk 3]
```diff
@@ -39,3 +39,53 @@ container_test(
     configs = ["//tests/docker/configs:deb_image_with_dpkgs.yaml"],
     image = ":deb_image_with_dpkgs",
 )
+
+container_image(
```
Comment (nlopezgi) [inline tests/docker/BUILD]: lets improve naming
remove image_with_ from all targets here here

File: tests/docker/BUILD [hunk 4]
```diff
@@ -39,3 +39,53 @@ container_test(
...
+    name = "image_with_some_cmd_entrypoint",
```
Comment (nlopezgi) [inline tests/docker/BUILD]: set_cmd_and_entrypoint

also add tests with just cmd and just entrypoint

File: tests/docker/BUILD [hunk 5]
```diff
@@ -39,3 +39,53 @@ container_test(
...
+    base = "@distroless_base//image",
+    cmd = ["foo"],
+    entrypoint = ["bar"],
+)
+
+container_test(
+    name = "image_with_some_cmd_entrypoint_test",
```
Comment (nlopezgi) [inline tests/docker/BUILD]: set_cmd_and_entrypoint_test

File: tests/docker/BUILD [hunk 6]
```diff
@@ -39,3 +39,53 @@ container_test(
...
+    configs = ["//tests/docker/configs:image_with_some_cmd_entrypoint.yaml"],
+    image = ":image_with_some_cmd_entrypoint",
+)
+
+container_image(
+    name = "image_should_still_have_cmd_entrypoint",
```
Comment (nlopezgi) [inline tests/docker/BUILD]: base_with_cmd_and_entry

File: tests/docker/BUILD [hunk 7]
```diff
@@ -39,3 +39,53 @@ container_test(
...
+)
+
+container_test(
+    name = "image_should_still_have_cmd_entrypoint_test",
+    configs = ["//tests/docker/configs:image_with_some_cmd_entrypoint.yaml"],
+    image = ":image_should_still_have_cmd_entrypoint",
+)
+
+container_image(
+    name = "image_with_null_cmd_entrypoint",
```
Comment (nlopezgi) [inline tests/docker/BUILD]: null_entrypoint_none

also add:
null_entrypoint_empty_str
null_entrypoint_empty_list

File: tests/docker/BUILD (hunk 3 above)
Comment (xingao267) [reply tests/docker/BUILD]: Done.

File: tests/docker/BUILD (hunk 4 above)
Comment (xingao267) [reply tests/docker/BUILD]: Done.

File: tests/docker/BUILD (hunk 5 above)
Comment (xingao267) [reply tests/docker/BUILD]: Done.

File: tests/docker/BUILD (hunk 7 above)
Comment (xingao267) [reply tests/docker/BUILD]: Done.

File: tests/docker/BUILD (hunk 6 above)
Comment (xingao267) [reply tests/docker/BUILD]: Done.

File: tests/docker/BUILD (hunk 2 above)
Comment (xingao267) [reply tests/docker/BUILD]: Done.

File: container/image.bzl (hunk 1 above)
Comment (xingao267) [reply container/image.bzl]: Done.

**Pull request merged**